File: c++/src/4.cpp
#include <stdio.h>
#include <math.h>
#include <string>
#include <iostream>
#include <algorithm>

using namespace std;

int isPalindome(int num){
    string tmp = to_string(num);
    string tmp2 = tmp;
    reverse(tmp2.begin(), tmp2.end());
    return tmp == tmp2;
}

int main(int argc, char *argv[])
{
    int max_palindrome = 0;
    int test;
    for (int i = 1; i < 1000; i++) {
        for (int j = 1; j < 1000; j++) {
            test = i*j;
            if (isPalindome(test)) {
                if (test > max_palindrome)
                    max_palindrome = test;
            }

        }
    }

    cout << max_palindrome << "\n";
    return 0;
}

File: java/src/4.java
class problem4 {
    public static void main(String[] args){
        int max_palindrome = 0;
        int test;
        for (int i = 1; i < 1000; i++) {
            for (int j = 0; j < 1000; j++) {
                test = i*j;
                if (isPalindrome(test)) {
                    if (test > max_palindrome) {
                        max_palindrome = test;
                    }
                }
            }
        }
        System.out.println(String.valueOf(max_palindrome));
    }

    public static boolean isPalindrome(int n) {
        String tmp = String.valueOf(n);
        String reverse = new StringBuilder(tmp).reverse().toString();
        return tmp.equals(reverse);
    }
}
